src/test/java/io/managed/services/test/cluster/DataPlaneClusterTest.java

```diff
@@ -0,0 +1,359 @@
+            try {
+                KafkaMgmtApiUtils.deleteKafkaByNameIfExists(kafkaMgmtApi, KAFKA_INSTANCE_NAME);
+                log.info("wait until kafka is deleted");
+                KafkaMgmtApiUtils.waitUntilKafkaIsDeleted(kafkaMgmtApi, kafkaRequest.getId());
+            } catch (Exception e) {
+                log.error("error while cleaning kafka instance: %s", e);
+            }
+        }
+    }
+
+    @Test()
+    @SneakyThrows
+    public void testStandardKafkaNodeAutoscaling() {
+
+        // make sure dummy instances are cleaned.
+        KafkaMgmtApiUtils.deleteSearchedKafkaInstancesByOwner(kafkaMgmtApi, "jano", Environment.PRIMARY_USERNAME);
+
+        //take a look on machine sets with name (INITIAL)
+        Map<String, Integer> machineSetToReadyNodeCountSnapshotBefore =  FleetshardUtils.getReadyNodesPerEachMachineSetContainingName(oc, "kafka-standard");
+        log.info(machineSetToReadyNodeCountSnapshotBefore);
+
+        // obtain info how many dev instances there are (including reserved deployment)
+        int remainingStandardInstancesCapacity = FleetshardUtils.getCapacityRemainingUnitsFromMKAgent(oc, ManagedKafkaType.standard);
+        log.info("currently remaining standard managed kafka instances capacity '{}'", remainingStandardInstancesCapacity);
+
+        // there is not enough space in the cluster to observe nodes scaling.
+        if (remainingStandardInstancesCapacity == 0)
+            throw new SkipException("Already reached max limit of standard instances created");
+
+        // creating one instance less than to reach max as that would not help observing max node creation anyway.
+        log.info("creating at least '{}' standard managed kafka instances", remainingStandardInstancesCapacity);
+
+        // wait up to 15 minutes for observing increase in nodes number in relevant machine sets, or creation of new instance
+        AtomicInteger placeholderInstanceSuffixCounter = new AtomicInteger(1);
+
+        try {
+            TestUtils.waitFor(
+                "update of number of nodes scaled due to creation of new standard managed kafka instance",
+                Duration.ofSeconds(10),
+                Duration.ofMinutes(15),
+                ready -> {
+                    int remainingCapacity = FleetshardUtils.getCapacityRemainingUnitsFromMKAgent(oc, ManagedKafkaType.standard);
+                    log.info("remaining capacity in the cluster '{}'", remainingCapacity);
+
+                    // spawn new instance if there is a capacity for it
+                    int minNodeInMachineSet =  machineSetToReadyNodeCountSnapshotBefore.values().stream().reduce(0, Integer::min);
+
+                    // if we did not reach max capacity, and at least of machineSet was not already at highest number we attempt creating new instances (quite common occurrence as downscaling take much longer time)
+                    if (remainingCapacity > 0 && minNodeInMachineSet < maxDeveloperInstancesInCluster) {
```

Comment:
```suggestion
                    if (remainingCapacity > 0 && minNodeInMachineSet < maxStandardInstancesInCluster) {
```

The test is testing the kafka-standard machine pool.

Reply:
thanks, yeah this really was logical mistake on my side. Fixed. 
just additionL  Because `developer` pool basically always has place (much more than `standard`), and we expect option of failing on quota it should only result in 2-3 more retrying on failed quota while waiting anyway.